src/python/examples/grpc_image_client.py

```
@@ -371,9 +376,14 @@ def requestGenerator(input_name, output_name, c, h, w, format, dtype, FLAGS,
                                                     version=FLAGS.model_version)
     config_response = grpc_stub.ModelConfig(config_request)
 
-    input_name, output_name, c, h, w, format, dtype = parse_model(
+    max_batch_size, input_name, output_name, c, h, w, format, dtype = parse_model(
         metadata_response, config_response.config)
 
+    supports_batching = max_batch_size > 0
+    if not supports_batching and FLAGS.batch_size != 1:
+        print("ERROR: This model doesn't support batching.")
```

raise Exception instead of print + exit

---

Updated